src/Objs/Homepage/SiteColorsPicker.tsx

```diff
@@ -60,15 +61,40 @@ function AdvancedColorPicker({
   color: string
   setColor: (color: string) => void
 }) {
-  const [iColor] = useColor(color)
+  const [iColor, setIColor] = useState(ColorService.convert('hex', color))
+
+  const [hexHsvCache, setHexHsvCache] = useState<
+    Record<
+      string,
+      {
+        readonly h: number
+        readonly s: number
+        readonly v: number
+        readonly a: number
+      }
+    >
+  >({})
+
+  useEffect(() => {
+    const cachedHsv = hexHsvCache[color]
+    const newIColor = cachedHsv
+      ? ColorService.convert('hsv', cachedHsv)
+      : ColorService.convert('hex', color)
+
+    setIColor(newIColor)
+  }, [color, hexHsvCache])
 
   return (
     <ColorPicker
       height={100}
       color={iColor}
       hideInput={['hsv']}
       hideAlpha={true}
-      onChange={(newIColor) => setColor(newIColor.hex)}
+      onChange={(newIColor) => {
+        const newColor = newIColor.hex
+        setHexHsvCache((prev) => ({ ...prev, [newColor]: newIColor.hsv }))
```

Comment:
* [ ] Why do we need to cache more than the current color? If I remove `...prev,` it appears to work nicely. Can we simplify?

I did this quick prototype:
```tsx
{
  const [iColor] = useColor(color)
  const [hiResIColor, setHiResIColor] = useState(iColor)

  return (
    <ColorPicker
      height={100}
      color={hiResIColor.hex === iColor.hex ? hiResIColor : iColor}
      hideInput={['hsv']}
      hideAlpha={true}
      onChange={(newIColor) => {
        setHiResIColor(newIColor)
        setColor(newIColor.hex)
      }}
    />
  )
}
```

Reply:
Nice simplification!